pkg/internal/checkup/checkup_test.go

```diff
@@ -41,3 +51,57 @@ func TestCheckupShouldSucceed(t *testing.T) {
 
 	assert.Equal(t, expectedResults, actualResults)
 }
+
+func TestSetupShouldFail(t *testing.T) {
+	t.Run("when VMI creation fails", func(t *testing.T) {
+		expectedVMICreationFailure := errors.New("failed to create VMI")
+
+		testClient := newClientStub()
+		testClient.vmiCreationFailure = expectedVMICreationFailure
+		testCheckup := checkup.New(testClient, testNamespace, newTestConfig())
+
+		assert.ErrorContains(t, testCheckup.Setup(context.Background()), expectedVMICreationFailure.Error())
+	})
+}
+
+type clientStub struct {
+	createdVMIs        map[string]*kvcorev1.VirtualMachineInstance
+	vmiCreationFailure error
+}
+
+func newClientStub() *clientStub {
+	return &clientStub{
+		createdVMIs: map[string]*kvcorev1.VirtualMachineInstance{},
+	}
+}
+
+func (cs *clientStub) CreateVirtualMachineInstance(_ context.Context,
+	namespace string,
+	vmi *kvcorev1.VirtualMachineInstance) (*kvcorev1.VirtualMachineInstance, error) {
+	if cs.vmiCreationFailure != nil {
+		return nil, cs.vmiCreationFailure
+	}
+
+	vmiFullName := fmt.Sprintf("%s/%s", namespace, vmi.Name)
+	cs.createdVMIs[vmiFullName] = vmi
+
+	return vmi, nil
+}
+
+func newTestConfig() config.Config {
+	return config.Config{
+		PodName:                           "",
+		PodUID:                            "",
+		NUMASocket:                        0,
+		NetworkAttachmentDefinitionName:   "",
+		TrafficGeneratorNodeLabelSelector: "",
+		DPDKNodeLabelSelector:             "",
+		TrafficGeneratorPacketsPerSecondInMillions: 0,
+		PortBandwidthGB:                0,
+		TrafficGeneratorEastMacAddress: nil,
+		TrafficGeneratorWestMacAddress: nil,
+		DPDKEastMacAddress:             nil,
+		DPDKWestMacAddress:             nil,
+		TestDuration:                   0,
```

Comment:
We need to add relevant configuration.

Reply:
why not add only the mandatory ones?

Reply:
In this stage, the configuration should be filled with all the user-supplied and default values.